docs/path-resolution.rst

```diff
@@ -0,0 +1,465 @@
+   ``urls`` one by one, until one is loaded successfully or the end of the list is reached.
+
+   The source unit names are determined the same way as when using ``content`` - they are keys of
+   the ``sources`` dictionary and the content of ``urls`` does not affect them in any way.
+
+   .. index:: standard input, stdin, <stdin>
+
+#. **Standard input**
+
+   On the command line it is also possible to provide the source is by sending it to compiler's
+   standard input:
+
+   .. code-block:: bash
+
+       echo 'import "./util.sol"; contract C {}' | solc -
+
+   ``-`` used as one of the arguments instructs the compiler to place the content of the standard
+   input in the virtual filesystem under a special source unit name: ``<stdin>``.
+
+Once the VFS is initialized, additional files can still be added to it only through the import
+callback.
+
+.. index:: ! import; path
+
+Imports
+=======
+
+The import statement specifies an *import path*, which, after being lightly processed, becomes a
+source unit name.
+Based on how the processing is performed, we can divide imports into two categories:
+
+- :ref:`Direct imports <direct-imports>`, where you specify the full source unit name.
+- :ref:`Relative imports <relative-imports>`, where you specify a path to be combined with the
+  source unit name of the importing file.
+
+.. index:: ! direct import, import; direct
+.. _direct-imports:
+
+Direct Imports
+--------------
+
+An import that does not start with ``./`` or ``../`` is a *direct import*.
+
+::
+
+    import "/project/lib/util.sol";         // source unit name: /project/lib/util.sol
+    import "lib/util.sol";                  // source unit name: lib/util.sol
+    import "@openzeppelin/address.sol";     // source unit name: @openzeppelin/address.sol
+    import "https://example.com/token.sol"; // source unit name: https://example.com/token.sol
+
+After applying any :ref:`import remappings <import-remapping>` the import path simply becomes the
+source unit name.
+
+.. note::
+
+    A source unit name is just an identifier and even if its value happens to look like a path, it
+    is not subject to the normalization rules you would typically expect from filesystem paths.
+    Any ``/./`` or ``/../`` seguments or sequences of multiple slashes remain a part of it.
+    When the source is provided via Standard JSON interface it is entirely possible to associate
+    different content with source unit names that would refer to the same file on disk.
+
+When the source is not available in the virtual filesystem, the compiler passes the source unit name
+to the import callback.
+The Host Filesystem Loader will attempt to use it as a path and look up the file on disk.
+At this point the platform-specific normalization rules kick in and ``/project/lib/math.sol`` and
+``/project/lib/../lib///math.sol`` may actually result in the same file being loaded.
+Note, however, that the compiler will still see them as separate source units that just happen to
```

Comment:
Maybe we could put this also in a `.. note::` section?

Reply:
I tried to change it but left it as is after all. This sentence is strongly connected to this paragraph and I think that putting in a note (or even just in a separate paragraph) makes it read weird.

Reply:
It's rather confusing because the two import paths somehow appear out of nowhere and it's not clear what they refer to.

Reply:
ok, now I get what the problem is. I have reworded that paragraph and changed how the examples are introduced so that now it makes sense as an actual note.